### PR TITLE
added test cases for exporting collection items like dict, list

### PR DESCRIPTION
**Pull request by aufziehvogel**
Test cases for issue #532, which has already been fixed some time ago.

### CODE REVIEW

**Comment by kmike** [comment]
:+1:

File: scrapy/contrib/exporter/__init__.py [hunk 1]
```diff
@@ -154,6 +154,7 @@ def _export_xml_field(self, name, serialized_value):
     if sys.version_info[:3] >= (2, 7, 4):
         def _xg_characters(self, serialized_value):
             if not isinstance(serialized_value, unicode):
+                serialized_value = str(serialized_value)
```
Comment (nramirezuy) [inline scrapy/contrib/exporter/__init__.py]: Can serialized value be a non-ascii string ?

Comment (kmike) [inline scrapy/contrib/exporter/__init__.py]: If it is a non-ascii byte string it will be decoded to unicode using self.encoding. `str(serialized_value)` will do nothing in this case. If the string is unicode `str(serialized_value)` won't be called (see the condition above).

Comment (nramirezuy) [inline scrapy/contrib/exporter/__init__.py]: You are right! But what about unicode inside lists or dicts.

Comment (kmike) [inline scrapy/contrib/exporter/__init__.py]: From the source code it seems they'll work fine - _export_xml_field method calls itself recursively, so _xg_characters method shouldn't receive containers (lists or dicts). 

A test for this would be nice though.

**Comment by nramirezuy** [comment]
@aufziehvogel can you add a test with unicode inside list and dict objects?

**Comment by aufziehvogel** [comment]
@nramirezuy @kmike Sure I can, just give me some time :) Somewhen this week, I'd say.

**Comment by kmike** [comment]
Thanks @aufziehvogel!

**Comment by aufziehvogel** [comment]
Ok, there are one version I would think is correct and one version that passes all tests.

My problem is: We have this function `_to_str_if_unicode` sometimes being called within `serialize_field`, which in turn is called in `_get_serialized_fields`.

In the `PythonItemExporter` the unicode is transformed to string, as expected by the function name. But in `JsonLinesItemExporterTest` the unicode remains.

What is the expected output? I guess it should always be string, not unicode? Otherwise we probably would not have this function.

Then we need to find out, why in `JsonLinesItemExporterTest` it is unicode coming out. I guess it has to do with the line being `serializer = field.get('serializer', self._to_str_if_unicode)`, so we only get `_to_str_if_unicode` if there is no other serializer? And is there some other default serializer for JSON?

**Comment by kmike** [comment]
@aufziehvogel see https://github.com/scrapy/scrapy/issues/1080 - I think we should kill _to_str_if_unicode.

**Comment by aufziehvogel** [comment]
@kmike This sounds reasonable, but I'm thinking about the split between both pull requests. I mean, I can do both, I'm a bit into the code now (not much, but a bit).

Then I just have to get this PR "right" (right meaning = right for current situation, even if the other PR is rejected) and the question is, can we see it as correct, that atm we get unicode for JSON exporters and string for Python exporter?...

**Comment by kmike** [comment]
@aufziehvogel maybe you can make this PR test-only, without `serialized_value = str(serialized_value)` line? Fixing unicode issues will be left for #1080.

**Comment by aufziehvogel** [comment]
@kmike of course I can, but I still need to know which output to expect :) Because it's different if I set expected value to `u"井上"` or to `"井上"`. Both work, but both get different representations in the test case (and the output of the test case).

One is `u'\u4e95\u4e0a'` and one is `'\xe4\xba\x95\xe4\xb8\x8a'`

**Comment by kmike** [comment]
Item exporters work with unicode since https://github.com/scrapy/scrapy/pull/1499.
